### PR TITLE
NET-4858 - xds v2 - implement base connect proxy functionality for routes

### PR DESCRIPTION
**Pull request by jmurret**
Description
This PR is aimed at single cluster connect proxy tests for routes. It is the first phase of tests and will not include functionality / tests for:

JWT auth
escape hatch clusters
gateways - terminating, ingress, mesh, api
cluster peering

PR Checklist
 updated test coverage

### CODE REVIEW

File: agent/xds/naming/naming.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Copyright (c) HashiCorp, Inc.
-// SPDX-License-Identifier: BUSL-1.1
+// SPDX-License-Identifier: MPL-2.0
```
Comment (DanStough) [inline agent/xds/naming/naming.go]: Same weird licensing edit.

File: agent/xdsv2/cluster_resources.go [hunk 2]
```diff
@@ -325,6 +342,19 @@ func addEnvoyLBToCluster(dynamicConfig *pbproxystate.DynamicEndpointGroupConfig,
...
+			SocketAddress: &envoy_core_v3.SocketAddress{
+				Address: ip,
+				PortSpecifier: &envoy_core_v3.SocketAddress_PortValue{
+					PortValue: port,
+				},
+			},
+		},
+	}
+}
+
```
Comment (DanStough) [inline agent/xdsv2/cluster_resources.go:357]: Is this also somewhere else, like endpoints?

**Review by DanStough — APPROVED** [review]
A few minor things you'll want to inspect before merging.

File: agent/xds/listeners_test.go [hunk 3]
```diff
@@ -36,7 +37,7 @@ type listenerTestCase struct {
 	// test input.
 	overrideGoldenName string
 	generatorSetup     func(*ResourceGenerator)
-	alsoRunTestForV2   bool
```
[the remark this answers is not in the document]
Comment (jmurret) [reply agent/xds/listeners_test.go]: I just fixed that.

File: agent/xds/naming/naming.go (hunk 1 above)
Comment (jmurret) [reply agent/xds/naming/naming.go]: thank you.  just fixed that as well.

**Pull request merged**